// solutionsRegistry/test/SolutionsRegistryTests.js
/*
 * Solutions Registry Tests
 *
 * Copyright 2016 Raising the Floor - U.S.
 *
 * Licensed under the New BSD license. You may not use this file except in
 * compliance with this License.
 *
 * You may obtain a copy of the License at
 * https://github.com/GPII/universal/blob/master/LICENSE.txt
 */

"use strict";

var fluid = require("infusion"),
    gpii = fluid.registerNamespace("gpii"),
    jqUnit = fluid.registerNamespace("jqUnit"),
    kettle = require("kettle");

fluid.registerNamespace("gpii.tests");

kettle.loadTestingSupport();

require("solutionsRegistry");

gpii.tests.SolutionRegistryJawsVersions = function () {
    console.log("First test!");
    jqUnit.assertDeepEq("One", 1, 1);
    jqUnit.assertDeepEq("Two", 2, 2);
    var solreg = gpii.solutionsRegistry();
    solreg.assemble();
};

fluid.defaults("gpii.tests.SolutionsRegistry", {
    gradeNames: ["fluid.test.testEnvironment"],
    components: {
        tester: {
            type: "gpii.tests.SolutionRegistryTestCaseHolder"
        }
    }
});

fluid.defaults("gpii.tests.SolutionRegistryTestCaseHolder", {
    gradeNames: ["fluid.test.testCaseHolder"],
    modules: [{
        name: "SolutionsRegistryInheritance",
        tests: [{
            expect: 2,
            name: "SolutionRegistryJawsVersions",
            func: "gpii.tests.SolutionRegistryJawsVersions"
        }
        ]}
    ]
});

module.exports = kettle.test.bootstrap("gpii.tests.SolutionsRegistry");
